atomicfu/build.gradle

```diff
@@ -1,386 +0,0 @@
-
-    // Always compile kotlin classes before the module descriptor.
-    dependsOn(compileKotlinTask)
-
-    // Add the module-info source file.
-    source(sourceDir)
-
-    // Also add the module-info.java source file to the Kotlin compile task.
-    // The Kotlin compiler will parse and check module dependencies,
-    // but it currently won't compile to a module-info.class file.
-    // Note that module checking only works on JDK 9+,
-    // because the JDK built-in base modules are not available in earlier versions.
-    def javaVersion = compileKotlinTask.kotlinJavaToolchain.javaVersion.getOrNull()
-    if (javaVersion?.isJava9Compatible() == true) {
-        logger.info("Module-info checking is enabled; $compileKotlinJvm is compiled using Java $javaVersion")
-        compileKotlinJvm.source(sourceDir)
-    } else {
-        logger.info("Module-info checking is disabled")
-    }
-
-    // Set the task outputs and destination dir
-    outputs.dir(targetDir)
-    destinationDirectory.set(targetDir)
-
-    // Configure JVM compatibility
-    sourceCompatibility = JavaVersion.VERSION_1_9.toString()
-    targetCompatibility = JavaVersion.VERSION_1_9.toString()
-
-    // Set the Java release version.
-    options.release.set(9)
-
-    // Ignore warnings about using 'requires transitive' on automatic modules.
-    // not needed when compiling with recent JDKs, e.g. 17
-    options.compilerArgs.add("-Xlint:-requires-transitive-automatic")
-
-    // Patch the compileKotlinJvm output classes into the compilation so exporting packages works correctly.
-    options.compilerArgs.addAll(["--patch-module", "$moduleName=${compileKotlinTask.destinationDirectory.get().getAsFile()}"])
-
-    // Use the classpath of the compileKotlinJvm task.
-    // Also ensure that the module path is used instead of classpath.
-    classpath = compileKotlinTask.libraries
-    modularity.inferModulePath.set(true)
-
-    doFirst {
-        logger.warn("Task destination directory: ${destinationDirectory.get().getAsFile()}")
-    }
-}
-
-// Configure the JAR task so that it will include the compiled module-info class file.
-tasks.named("jvmJar") {
-    manifest {
-        attributes(["Multi-Release": true])
-    }
-    from(compileJavaModuleInfo.destinationDirectory) {
-        into("META-INF/versions/9/")
-    }
-}
-
-jvmTest {
-    exclude "**/AtomicfuBytecodeTest*", "**/AtomicfuReferenceJsTest*", '**/TopLevelGeneratedDeclarationsReflectionTest.*', '**/SyntheticFUFieldsTest.*'
-    // run them only for transformed code
-}
-
-jvmTest.dependsOn jvmTestAll
-
-tasks.matching { it.name == "generatePomFileForKotlinMultiplatformPublication" }.configureEach {
-    dependsOn(tasks["generatePomFileForJvmPublication"])
-}
-
-// Workaround for https://youtrack.jetbrains.com/issue/KT-58303:
-// the `clean` task can't delete the expanded.lock file on Windows as it's still held by Gradle, failing the build
-tasks.clean {
```

Comment:
removed this block, because since KGP `2.0.0` we download Kotlin Native on execution phase